snuba/utils/streams/consumer.py

```diff
@@ -80,6 +84,26 @@ class KafkaPayload:
     value: bytes
 
 
+def as_kafka_configuration_bool(value: Any) -> bool:
+    """
+    Convert a value to a Python boolean, using the same conversions used by
+    the librdkafka configuration parser.
+    """
+    # https://github.com/edenhill/librdkafka/blob/c8293af/src/rdkafka_conf.c#L1633-L1660
+    if isinstance(value, str):
```

Comment:
can this be an integer 1 / 0 ? (please say no)

Reply:
```python
>>> c = Consumer({'bootstrap.servers': '127.0.0.1:9092', 'group.id': 'asdf', 'enable.auto.commit': 1})
>>> c = Consumer({'bootstrap.servers': '127.0.0.1:9092', 'group.id': 'asdf', 'enable.auto.commit': '1'})
>>> c = Consumer({'bootstrap.servers': '127.0.0.1:9092', 'group.id': 'asdf', 'enable.auto.commit': 1.0})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Expected bool value for "enable.auto.commit": true or false"}
>>> c = Consumer({'bootstrap.servers': '127.0.0.1:9092', 'group.id': 'asdf', 'enable.auto.commit': 'bad'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Expected bool value for "enable.auto.commit": true or false"}
>>> c = Consumer({'bootstrap.servers': '127.0.0.1:9092', 'group.id': 'asdf', 'enable.auto.commit': '1  '})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Expected bool value for "enable.auto.commit": true or false"}
```

🙃  (The last one is my favorite.)

I'll try to track down where that conversion happens…

Reply:
Fixed this one here: e978e1b38c1d394215afad17f10765b59f35c07d